server/index.js

```diff
@@ -12,58 +13,66 @@ app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.json());
 
 app.get('/traits/:product_id', (req, res) => {
-  const id = req.params.product_id;
+  const id = +req.params.product_id;
   const traits = fetchers.fetchTraitsForProduct(id);
+
   traits.then((traitsData) => {
     const traitProducts = traitsData.traits.reduce((acc, trait) => {
       return acc.concat(fetchers.fetchProductsForTrait(trait, id));
     }, []);
     Promise.all(traitProducts)
       .then((resultsFinal) => {
+        // console.log('-----rFinal', resultsFinal);
         resultsFinal.forEach((result) => {
+          if (result.products.indexOf(id) >= 0) {
+            const filteredProducts = result.products.filter((product) => {
+              return product !== id;
+            });
+            // eslint-disable-next-line no-param-reassign
+            result.products = filteredProducts;
+          }
```

Comment:
I think lines 27 - 33 can be simplified to `result.products = result.products.filter(product => product !== id)`

Reply:
yes, although the lint rules complain. I'll change it

Reply:
Agreed, thank you! changed